avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1

```diff
@@ -113,35 +121,58 @@ Describe 'File/folder tests' -Tag 'Modules' {
       $pathExisting | Should -Be $true
     }
 
-    It '[<moduleFolderName>] Module should contain a [` tests/waf-aligned `] folder.' -TestCases ($moduleFolderTestCases | Where-Object { $_.isTopLevelModule }) {
+    It '[<moduleFolderName>] Module should contain a [` tests/e2e/*waf-aligned `] folder.' -TestCases $topLevelModuleTestCases {
 
       param(
         [string] $moduleFolderPath
       )
 
-      $pathExisting = Test-Path (Join-Path -Path $moduleFolderPath 'tests' 'waf-aligned')
-      $pathExisting | Should -Be $true
+      $wafAlignedFolder = Get-ChildItem -Directory (Join-Path -Path $moduleFolderPath 'tests' 'e2e') -Filter '*waf-aligned'
+      $wafAlignedFolder | Should -Not -BeNullOrEmpty
     }
 
-    It '[<moduleFolderName>] Module should contain a [` min `] folder.' -TestCases ($moduleFolderTestCases | Where-Object { $_.isTopLevelModule }) {
+    It '[<moduleFolderName>] Module should contain a [` tests/e2e/*defaults `] folder.' -TestCases $topLevelModuleTestCases {
 
       param(
         [string] $moduleFolderPath
       )
 
-      $pathExisting = Test-Path (Join-Path -Path $moduleFolderPath 'tests' 'min')
-      $pathExisting | Should -Be $true
+      $defaultsFolder = Get-ChildItem -Directory (Join-Path -Path $moduleFolderPath 'tests' 'e2e') -Filter '*defaults'
+      $defaultsFolder | Should -Not -BeNullOrEmpty
     }
 
-    It '[<moduleFolderName>] Folder should contain one or more test files.' -TestCases $folderTestCases {
+    It '[<moduleFolderName>] Module should contain one [` main.test.bicep `] file in each e2e test folder.' -TestCases $topLevelModuleTestCases {
 
       param(
         [string] $moduleFolderName,
         [string] $moduleFolderPath
       )
 
-      $moduleTestFilePaths = Get-ModuleTestFileList -ModulePath $moduleFolderPath | ForEach-Object { Join-Path $moduleFolderPath $_ }
-      $moduleTestFilePaths.Count | Should -BeGreaterThan 0
+      $e2eTestFolderPathList = Get-ChildItem -Directory (Join-Path -Path $moduleFolderPath 'tests' 'e2e')
+      foreach ($e2eTestFolderPath in $e2eTestFolderPathList) {
+        $pathExisting = Test-Path (Join-Path -Path $e2eTestFolderPath 'main.test.bicep')
+        $pathExisting | Should -Be $true
+      }
+    }
+
+    It '[<moduleFolderName>] Module should contain a [` tests/unit `] folder.' -TestCases $topLevelModuleTestCases {
+
+      param(
+        [string] $moduleFolderPath
+      )
+
+      $pathExisting = Test-Path (Join-Path -Path $moduleFolderPath 'tests' 'unit')
+      $pathExisting | Should -Be $true
+    }
+
+    It '[<moduleFolderName>] Module should contain a [` tests/unit/custom.tests.ps1 `] file.' -TestCases $topLevelModuleTestCases {
```

Comment:
That's actually not a requirement. 'Any' Pester test file in the Unit folder would be picked up :) The name doesn't matter. One could argue though - if the test folder exists, it should contain at least 1 `.test.ps1` file

Reply:
Removed